ga4gh/datamodel/reads.py

```diff
@@ -189,6 +191,34 @@ def toProtocolElement(self):
         stats.baseCount = None  # TODO requires iterating through all reads
         readGroup.stats = stats
         readGroup.programs = self.getPrograms()
+        experiment = protocol.Experiment()
+        # required experiment fields
+        experiment.id = 'TODO'
+        experiment.instrumentModel = 'TODO'
+        experiment.recordCreateTime = 'TODO'
+        experiment.recordUpdateTime = 'TODO'
+        experiment.sequencingCenter = 'TODO'
+        samFile = self.getFileHandle(self._samFilePath)
+        if 'RG' in samFile.header:
+            # TODO change after merge
+            readGroupHeader = samFile.header['RG'][0]
+            experiment.description = readGroupHeader.get('DS', None)
+            experiment.id = 'TODO'
+            experiment.info = {}
+            experiment.instrumentDataFile = 'TODO'
```

Comment:
If a field is nullable, we should set it to ``None`` rather than TODO. I'd guess that some of these will never be done.